src/api/uploads/BaseUpload.js

```diff
@@ -60,11 +69,51 @@ class BaseUpload extends Base {
         this.xhr.options({
             url,
             data: attributes,
-            successHandler: this.preflightSuccessHandler,
+            successHandler: this.preflightSuccessReachabilityHandler,
             errorHandler: this.preflightErrorHandler,
         });
     };
 
+    /**
+     * Handles successful preflight response.
+     * Performs a upload reachability test before calling preflightSuccessHandler.
+     *
+     * @param {Object} - Request options
+     * @return {Promise} Async function promise
+     */
+    preflightSuccessReachabilityHandler = async ({ data }: { data: { upload_url?: string } }): Promise<any> => {
+        if (this.isDestroyed()) {
+            return;
+        }
+
+        const uploadUrl = data.upload_url;
+        // If uploadUrl is not avaliable, don't make reachability test
+        if (!uploadUrl) {
+            this.preflightSuccessHandler({ data });
```

Comment:
why wouldn't we want to set the upload url here similar to how it's done in preflightSuccessHandler to determine if the upload host is reachable before continuing on with the upload?